src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -370,11 +370,7 @@ private class TemplateBindingContext
 
                 if (_defaults != null)
                 {
-                    _filters = new RouteValueDictionary(defaults);
-                    foreach (var kvp in _defaults)
-                    {
-                        _filters.Add(kvp.Key, kvp.Value);
-                    }
+                    _filters = new RouteValueDictionary(_defaults);
```

Comment:
- That change is necessary as we would end up trying to add duplicate entries(due to the fix of constructor for read-only dictionary that this PR handles)...good catch regarding the `TemplateBindingContext.DebuggerToString()`..will fix it..
- I do not see much value in making this change...will leave it as it is..